Replace std::enable_if with requires/concepts (part 1)

weinig opened the pull request:
#### 46496e5fd6f7ddf79ae435b18f4c815dc6e338a5
<pre>
Replace std::enable_if with requires/concepts (part 1)
<a href="https://bugs.webkit.org/show_bug.cgi?id=301805">https://bugs.webkit.org/show_bug.cgi?id=301805</a>

Reviewed by Darin Adler.

First pass of replacing uses of std::enable_if/std::enable_if_t with
the requires/concepts. This pass focuses on WTF and only does the
following trivial transformations:

  - std::is_same::value/std::is_same_v -&gt; std::same_as
  - std::is_integer::value/std::is_integer_v -&gt; std::integral
  - std::is_integer::value/std::is_integer_v + std::is_signed::value/std::is_signed_v -&gt; std::signed_integral
  - std::is_integer::value/std::is_integer_v + std::is_unsigned::value/std::is_unsigned_v -&gt; std::unsigned_integral
  - std::is_floating_point::value/std::is_floating_point_v -&gt; std::floating_point
  - IsSmartPtr::value -&gt; SmartPtr
  - IsSmartPtr::value + !IsSmartPtr::isNullable -&gt; NonNullableSmartPtr
  - std::is_*::value -&gt; std::is_*_v (e.g. std::is_function::value -&gt; std::is_function_v)
  - std::*::type -&gt; std::*_t (e.g. std::remove_pointer::type -&gt; std::remove_pointer_t)

Further adoption of stricter concepts, which might change the semantics,
such as std::is_base_of_v -&gt; std::derived_from, can be done in follow ups.

* Source/JavaScriptCore/assembler/CodeLocation.h:
* Source/JavaScriptCore/assembler/LinkBuffer.h:
* Source/WTF/wtf/ArgumentCoder.h:
* Source/WTF/wtf/CagedPtr.h:
* Source/WTF/wtf/CallbackAggregator.h:
* Source/WTF/wtf/CheckedArithmetic.h:
* Source/WTF/wtf/CodePtr.h:
* Source/WTF/wtf/CompactPointerTuple.h:
* Source/WTF/wtf/CompactPtr.h:
* Source/WTF/wtf/CompactUniquePtrTuple.h:
* Source/WTF/wtf/CompletionHandler.h:
* Source/WTF/wtf/EnumClassOperatorOverloads.h:
* Source/WTF/wtf/EnumTraits.h:
* Source/WTF/wtf/FastMalloc.h:
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Function.h:
* Source/WTF/wtf/FunctionTraits.h:
* Source/WTF/wtf/HashFunctions.h:
* Source/WTF/wtf/HashSet.h:
* Source/WTF/wtf/Hasher.h:
* Source/WTF/wtf/JSONValues.h:
* Source/WTF/wtf/ListHashSet.h:
* Source/WTF/wtf/Lock.h:
* Source/WTF/wtf/Locker.h:
* Source/WTF/wtf/Logger.h:
* Source/WTF/wtf/Packed.h:
* Source/WTF/wtf/PointerPreparations.h:
* Source/WTF/wtf/PtrTag.h:
* Source/WTF/wtf/RawPointer.h:
* Source/WTF/wtf/Ref.h:
* Source/WTF/wtf/SaturatedArithmetic.h:
* Source/WTF/wtf/SignedPtr.h:
* Source/WTF/wtf/StdLibExtras.h:
* Source/WTF/wtf/ThreadSafeWeakPtr.h:
* Source/WTF/wtf/Vector.h:
* Source/WTF/wtf/WeakRef.h:
* Source/WTF/wtf/cocoa/TypeCastsCocoa.h:
* Source/WTF/wtf/persistence/PersistentDecoder.h:
* Source/WTF/wtf/persistence/PersistentEncoder.h:
* Source/WTF/wtf/text/StringCommon.h:
* Source/WTF/wtf/text/StringConcatenate.h:
* Source/WTF/wtf/text/StringConcatenateNumbers.h:
* Source/WTF/wtf/text/StringImpl.h:
* Source/WTF/wtf/text/StringView.h:
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/applepay/PaymentMethod.h:
* Source/WebCore/Modules/indexeddb/IDBDatabaseIdentifier.h:
* Source/WebCore/Modules/indexeddb/IDBGetAllResult.h:
* Source/WebCore/Modules/indexeddb/shared/IDBCursorInfo.h:
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.h:
* Source/WebCore/Modules/indexeddb/shared/IDBOpenRequestData.h:
* Source/WebCore/Modules/indexeddb/shared/IDBRequestData.h:
* Source/WebCore/Modules/indexeddb/shared/IDBResourceIdentifier.h:
* Source/WebCore/Modules/indexeddb/shared/IDBResultData.h:
* Source/WebCore/Modules/speech/SpeechRecognitionUpdate.h:
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.h:
* Source/WebCore/bindings/js/SerializedScriptValue.h:
* Source/WebCore/editing/FontAttributeChanges.h:
* Source/WebCore/inspector/InspectorOverlayLabel.h:
* Source/WebCore/loader/COEPInheritenceViolationReportBody.h:
* Source/WebCore/loader/CORPViolationReportBody.h:
* Source/WebCore/loader/ResourceLoadStatistics.h:
* Source/WebCore/page/SecurityOrigin.h:
* Source/WebCore/platform/MediaSamplesBlock.h:
* Source/WebCore/platform/ProcessIdentity.h:
* Source/WebCore/platform/ProcessQualified.h:
* Source/WebCore/platform/RegistrableDomain.h:
* Source/WebCore/platform/ShareableResource.h:
* Source/WebCore/platform/SharedMemory.h:
* Source/WebCore/platform/ThreadSafeDataBuffer.h:
* Source/WebCore/platform/TrackInfo.h:
* Source/WebCore/platform/graphics/FontTaggedSettings.h:
* Source/WebCore/platform/graphics/GraphicsContextState.h:
* Source/WebCore/platform/graphics/LayoutRect.h:
* Source/WebCore/platform/graphics/PlatformDynamicRangeLimit.h:
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
* Source/WebCore/platform/graphics/Region.h:
* Source/WebCore/platform/graphics/ShareableBitmap.h:
* Source/WebCore/platform/graphics/cv/VideoFrameCV.h:
* Source/WebCore/platform/graphics/filters/DistantLightSource.h:
* Source/WebCore/platform/graphics/filters/FilterEffectGeometry.h:
* Source/WebCore/platform/graphics/filters/FilterOperations.h:
* Source/WebCore/platform/graphics/mac/AppKitControlSystemImage.h:
* Source/WebCore/platform/graphics/transforms/TransformOperations.h:
* Source/WebCore/platform/mediastream/MediaConstraints.h:
* Source/WebCore/platform/network/BlobPart.h:
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/SocketStreamError.h:
* Source/WebCore/rendering/EventRegion.h:
* Source/WebCore/svg/SVGPreserveAspectRatioValue.h:
* Source/WebCore/testing/MockContentFilterSettings.h:
* Source/WebGPU/WGSL/AST/ASTIdentifier.h:
* Source/WebGPU/WGSL/Types.h:
* Source/WebKit/Platform/IPC/ConnectionHandle.h:
* Source/WebKit/Platform/IPC/Decoder.h:
* Source/WebKit/Platform/IPC/Encoder.h:
* Source/WebKit/Platform/IPC/IPCEvent.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Scripts/generate-serializers.py:
* Source/WebKit/Shared/ApplePay/ApplePayPaymentSetupFeaturesWebKit.h:
* Source/WebKit/Shared/CallbackID.h:
* Source/WebKit/Shared/Cocoa/CoreIPCAVOutputContext.h:
* Source/WebKit/Shared/Cocoa/CoreIPCArray.h:
* Source/WebKit/Shared/Cocoa/CoreIPCCVPixelBufferRef.h:
* Source/WebKit/Shared/Cocoa/CoreIPCColor.h:
* Source/WebKit/Shared/Cocoa/CoreIPCContacts.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDateComponents.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSURLCredential.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSURLProtectionSpace.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPKDateComponentsRange.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPersonNameComponents.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPlistArray.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPlistDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPresentationIntent.h:
* Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.h:
* Source/WebKit/Shared/Cocoa/CoreIPCString.h:
* Source/WebKit/Shared/Cocoa/CoreIPCStringSet.h:
* Source/WebKit/Shared/Cocoa/CoreIPCURL.h:
* Source/WebKit/Shared/Databases/IndexedDB/WebIDBResult.h:
* Source/WebKit/Shared/MonotonicObjectIdentifier.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
* Source/WebKit/Shared/mac/SecItemRequestData.h:
* Source/bmalloc/bmalloc/Packed.h:
* Tools/TestWebKitAPI/Tests/WTF/MoveOnly.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:

Canonical link: <a href="https://commits.webkit.org/302456@main">https://commits.webkit.org/302456@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9588ccef9c36edb2603062a80033e8500e4f78aa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129139 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1398 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39975 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136518 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80523 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131010 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1330 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1275 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/98334 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66205 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132086 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1036 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115681 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/78982 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/128489 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/958 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33798 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79798 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/121130 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109405 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34294 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138992 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/127591 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1191 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1155 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/106868 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1243 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112017 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106702 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27161 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/980 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30541 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/53760 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1264 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64616 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/160606 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1090 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40091 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1135 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1188 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/ArgumentCoder.h:
```diff
@@ -34,7 +34,7 @@ namespace IPC {
 class Encoder;
 
-template<typename T, typename = void> struct ArgumentCoder;
+template<typename T> struct ArgumentCoder;
```
```suggestion
template<typename> struct ArgumentCoder;
```

darinadler commented on Source/WTF/wtf/CagedPtr.h:
```diff
@@ -79,8 +79,8 @@ class CagedPtr {
-    WTF_UNSAFE_BUFFER_USAGE typename std::enable_if<!std::is_same<void, U>::value, T>::type&
-    /* T& */ at(size_t index) const { return get()[index]; }
+        requires (!std::same_as<void, U>)
+WTF_UNSAFE_BUFFER_USAGE U& at(size_t index) const { return get()[index]; }
```
```suggestion
    WTF_UNSAFE_BUFFER_USAGE U& at(size_t index) const { return get()[index]; }
```

darinadler commented on Source/WTF/wtf/CheckedArithmetic.h:
```diff
@@ -139,27 +139,27 @@ class RecordOverflow {
-template <typename T> struct RemoveChecked;
-template <typename T> struct RemoveChecked<Checked<T>>;
+template<typename T, class OverflowHandler = CrashOnOverflow> class Checked;
+template<typename T> struct RemoveChecked;
```
```suggestion
template<typename, typename = CrashOnOverflow> class Checked;
template<typename> struct RemoveChecked;
```

darinadler commented on Source/WTF/wtf/SaturatedArithmetic.h:
```diff
@@ -39,9 +39,14 @@
 namespace WTF {
 
 // FIXME: Enhance this so it fails to compile calls where either of the arguments can be outside the range of the integral type instead of quietly converting.
```
Need a blank line after this so it's clearer the comment applies to all 3 functions, not just he first `saturatedSum`.

darinadler commented on Source/WTF/wtf/text/StringConcatenateNumbers.h:
```diff
@@ -51,8 +52,9 @@ class StringTypeAdapter<Integer, typename std::enable_if_t<std::is_integral_v<In
-using UnderlyingType = typename std::underlying_type_t<Enum>;
+    requires (std::is_enum_v<Enum>)
+class StringTypeAdapter<Enum> {
+    using UnderlyingType = typename std::underlying_type_t<Enum>;
```
```suggestion
private:
    using UnderlyingType = typename std::underlying_type_t<Enum>;
```

darinadler commented on Source/WebCore/platform/MediaSamplesBlock.h:
```diff
@@ -32,7 +32,7 @@
 
 namespace IPC {
-template<typename T, typename> struct ArgumentCoder;
+template<typename T> struct ArgumentCoder;
```
```suggestion
template<typename> struct ArgumentCoder;
```

darinadler commented on Source/WebCore/platform/TrackInfo.h:
```diff
@@ -36,7 +36,7 @@
 
 namespace IPC {
-template<typename T, typename> struct ArgumentCoder;
+template<typename T> struct ArgumentCoder;
```
```suggestion
template<typename> struct ArgumentCoder;
```

darinadler commented on Source/WebCore/platform/network/BlobPart.h:
```diff
@@ -29,14 +29,14 @@
 
 namespace IPC {
-template<typename T, typename> struct ArgumentCoder;
+template<typename T> struct ArgumentCoder;
```
```suggestion
template<typename> struct ArgumentCoder;
```

darinadler commented on Source/WebCore/svg/SVGPreserveAspectRatioValue.h:
```diff
@@ -24,7 +24,7 @@
 
 namespace IPC {
-template<typename T, typename> struct ArgumentCoder;
+template<typename T> struct ArgumentCoder;
```
```suggestion
template<typename> struct ArgumentCoder;
```

weinig committed:
Replace std::enable_if with requires/concepts (part 1)
https://bugs.webkit.org/show_bug.cgi?id=301805

Reviewed by Darin Adler.

First pass of replacing uses of std::enable_if/std::enable_if_t with
the requires/concepts. This pass focuses on WTF and only does the
following trivial transformations:

  - std::is_same::value/std::is_same_v -> std::same_as
  - std::is_integer::value/std::is_integer_v -> std::integral
  - std::is_integer::value/std::is_integer_v + std::is_signed::value/std::is_signed_v -> std::signed_integral
  - std::is_integer::value/std::is_integer_v + std::is_unsigned::value/std::is_unsigned_v -> std::unsigned_integral
  - std::is_floating_point::value/std::is_floating_point_v -> std::floating_point
  - IsSmartPtr::value -> SmartPtr
  - IsSmartPtr::value + !IsSmartPtr::isNullable -> NonNullableSmartPtr
  - std::is_*::value -> std::is_*_v (e.g. std::is_function::value -> std::is_function_v)
  - std::*::type -> std::*_t (e.g. std::remove_pointer::type -> std::remove_pointer_t)

Further adoption of stricter concepts, which might change the semantics,
such as std::is_base_of_v -> std::derived_from, can be done in follow ups.

* Source/JavaScriptCore/assembler/CodeLocation.h:
* Source/JavaScriptCore/assembler/LinkBuffer.h:
* Source/WTF/wtf/ArgumentCoder.h:
* Source/WTF/wtf/CagedPtr.h:
* Source/WTF/wtf/CallbackAggregator.h:
* Source/WTF/wtf/CheckedArithmetic.h:
* Source/WTF/wtf/CodePtr.h:
* Source/WTF/wtf/CompactPointerTuple.h:
* Source/WTF/wtf/CompactPtr.h:
* Source/WTF/wtf/CompactUniquePtrTuple.h:
* Source/WTF/wtf/CompletionHandler.h:
* Source/WTF/wtf/EnumClassOperatorOverloads.h:
* Source/WTF/wtf/EnumTraits.h:
* Source/WTF/wtf/FastMalloc.h:
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Function.h:
* Source/WTF/wtf/FunctionTraits.h:
* Source/WTF/wtf/HashFunctions.h:
* Source/WTF/wtf/HashSet.h:
* Source/WTF/wtf/Hasher.h:
* Source/WTF/wtf/JSONValues.h:
* Source/WTF/wtf/ListHashSet.h:
* Source/WTF/wtf/Lock.h:
* Source/WTF/wtf/Locker.h:
* Source/WTF/wtf/Logger.h:
* Source/WTF/wtf/Packed.h:
* Source/WTF/wtf/PointerPreparations.h:
* Source/WTF/wtf/PtrTag.h:
* Source/WTF/wtf/RawPointer.h:
* Source/WTF/wtf/Ref.h:
* Source/WTF/wtf/SaturatedArithmetic.h:
* Source/WTF/wtf/SignedPtr.h:
* Source/WTF/wtf/StdLibExtras.h:
* Source/WTF/wtf/ThreadSafeWeakPtr.h:
* Source/WTF/wtf/Vector.h:
* Source/WTF/wtf/WeakRef.h:
* Source/WTF/wtf/cocoa/TypeCastsCocoa.h:
* Source/WTF/wtf/persistence/PersistentDecoder.h:
* Source/WTF/wtf/persistence/PersistentEncoder.h:
* Source/WTF/wtf/text/StringCommon.h:
* Source/WTF/wtf/text/StringConcatenate.h:
* Source/WTF/wtf/text/StringConcatenateNumbers.h:
* Source/WTF/wtf/text/StringImpl.h:
* Source/WTF/wtf/text/StringView.h:
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/applepay/PaymentMethod.h:
* Source/WebCore/Modules/indexeddb/IDBDatabaseIdentifier.h:
* Source/WebCore/Modules/indexeddb/IDBGetAllResult.h:
* Source/WebCore/Modules/indexeddb/shared/IDBCursorInfo.h:
* Source/WebCore/Modules/indexeddb/shared/IDBDatabaseInfo.h:
* Source/WebCore/Modules/indexeddb/shared/IDBOpenRequestData.h:
* Source/WebCore/Modules/indexeddb/shared/IDBRequestData.h:
* Source/WebCore/Modules/indexeddb/shared/IDBResourceIdentifier.h:
* Source/WebCore/Modules/indexeddb/shared/IDBResultData.h:
* Source/WebCore/Modules/speech/SpeechRecognitionUpdate.h:
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.h:
* Source/WebCore/bindings/js/SerializedScriptValue.h:
* Source/WebCore/editing/FontAttributeChanges.h:
* Source/WebCore/inspector/InspectorOverlayLabel.h:
* Source/WebCore/loader/COEPInheritenceViolationReportBody.h:
* Source/WebCore/loader/CORPViolationReportBody.h:
* Source/WebCore/loader/ResourceLoadStatistics.h:
* Source/WebCore/page/SecurityOrigin.h:
* Source/WebCore/platform/MediaSamplesBlock.h:
* Source/WebCore/platform/ProcessIdentity.h:
* Source/WebCore/platform/ProcessQualified.h:
* Source/WebCore/platform/RegistrableDomain.h:
* Source/WebCore/platform/ShareableResource.h:
* Source/WebCore/platform/SharedMemory.h:
* Source/WebCore/platform/ThreadSafeDataBuffer.h:
* Source/WebCore/platform/TrackInfo.h:
* Source/WebCore/platform/graphics/FontTaggedSettings.h:
* Source/WebCore/platform/graphics/GraphicsContextState.h:
* Source/WebCore/platform/graphics/LayoutRect.h:
* Source/WebCore/platform/graphics/PlatformDynamicRangeLimit.h:
* Source/WebCore/platform/graphics/PlatformTimeRanges.h:
* Source/WebCore/platform/graphics/Region.h:
* Source/WebCore/platform/graphics/ShareableBitmap.h:
* Source/WebCore/platform/graphics/cv/VideoFrameCV.h:
* Source/WebCore/platform/graphics/filters/DistantLightSource.h:
* Source/WebCore/platform/graphics/filters/FilterEffectGeometry.h:
* Source/WebCore/platform/graphics/filters/FilterOperations.h:
* Source/WebCore/platform/graphics/mac/AppKitControlSystemImage.h:
* Source/WebCore/platform/graphics/transforms/TransformOperations.h:
* Source/WebCore/platform/mediastream/MediaConstraints.h:
* Source/WebCore/platform/network/BlobPart.h:
* Source/WebCore/platform/network/FormData.h:
* Source/WebCore/platform/network/SocketStreamError.h:
* Source/WebCore/rendering/EventRegion.h:
* Source/WebCore/svg/SVGPreserveAspectRatioValue.h:
* Source/WebCore/testing/MockContentFilterSettings.h:
* Source/WebGPU/WGSL/AST/ASTIdentifier.h:
* Source/WebGPU/WGSL/Types.h:
* Source/WebKit/Platform/IPC/ConnectionHandle.h:
* Source/WebKit/Platform/IPC/Decoder.h:
* Source/WebKit/Platform/IPC/Encoder.h:
* Source/WebKit/Platform/IPC/IPCEvent.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Scripts/generate-serializers.py:
* Source/WebKit/Shared/ApplePay/ApplePayPaymentSetupFeaturesWebKit.h:
* Source/WebKit/Shared/CallbackID.h:
* Source/WebKit/Shared/Cocoa/CoreIPCAVOutputContext.h:
* Source/WebKit/Shared/Cocoa/CoreIPCArray.h:
* Source/WebKit/Shared/Cocoa/CoreIPCCVPixelBufferRef.h:
* Source/WebKit/Shared/Cocoa/CoreIPCColor.h:
* Source/WebKit/Shared/Cocoa/CoreIPCContacts.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDateComponents.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSURLCredential.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSURLProtectionSpace.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPKDateComponentsRange.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPersonNameComponents.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPlistArray.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPlistDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCPresentationIntent.h:
* Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.h:
* Source/WebKit/Shared/Cocoa/CoreIPCString.h:
* Source/WebKit/Shared/Cocoa/CoreIPCStringSet.h:
* Source/WebKit/Shared/Cocoa/CoreIPCURL.h:
* Source/WebKit/Shared/Databases/IndexedDB/WebIDBResult.h:
* Source/WebKit/Shared/MonotonicObjectIdentifier.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
* Source/WebKit/Shared/mac/SecItemRequestData.h:
* Source/bmalloc/bmalloc/Packed.h:
* Tools/TestWebKitAPI/Tests/WTF/MoveOnly.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:

Canonical link: https://commits.webkit.org/302456@main
